ytypes/util_types.go

```diff
@@ -134,6 +210,134 @@ func StringToType(t reflect.Type, s string) (reflect.Value, error) {
 	return reflect.ValueOf(nil), fmt.Errorf("no matching type to cast for %v", t)
 }
 
+// stringToKeyType converts the given string to the type specified by the schema.
+// This is in contrast to StringToType, which requires foreknowledge of the
+// concrete type of the value. This is especially useful for converting the
+// string to a union type, where the final concrete value is not known.
+func stringToKeyType(schema *yang.Entry, parent interface{}, fieldName string, value string) (reflect.Value, error) {
+	ykind := schema.Type.Kind
+	switch ykind {
+	// TODO(wenbli): case yang.Yempty: case yang.Ybits: case yang.Ybool:
```

Comment:
`empty` cannot be a key type - so this doesn't need to be handled.

Should we add handling for `bool` here? (It's probably rare, but it'd be useful to handle and seems like it should be do-able.)

Reply:
I'll add the boolean representation in a follow-up PR. `KeyValueAsString` currently doesn't support it either.

Reply:
follow-up PR has been merged.